nixos/hardware/hetzner-cloud/default.nix

```diff
@@ -13,7 +13,7 @@
 
   config = {
     boot.growPartition = true;
-    boot.loader.grub.device = "/dev/sda";
+    boot.loader.grub.devices = lib.mkDefault [ "/dev/sda" ];
```

Comment:
This is a noop change because the grub module overrides this default:
https://github.com/NixOS/nixpkgs/blob/dd6c0ee128ae96cbb32855741ec85576a1368c1c/nixos/modules/system/boot/loader/grub/grub.nix#L715

That means that it works when Disko is being used but not otherwise.

I'm not sure what the best approach is to fix this. Make change the NixOS module?

Reply:
Should be fixed upstream to use lib.mkIf

Reply:
I would love to deprecate the singular version of grub.devices... It already has caused so much headache for me.